src/main/java/hudson/plugins/warnings/AggregatedWarningsResult.java

```diff
@@ -31,9 +32,30 @@
      *            the parsed result with all annotations
      * @param defaultEncoding
      *            the default encoding to be used when reading and parsing files
+     *
+     * @deprecated see {@link #AggregatedWarningsResult(Run, BuildHistory, ParserResult, String)}
      */
+    @Deprecated
     public AggregatedWarningsResult(final AbstractBuild<?, ?> build, final BuildHistory history, final ParserResult result,
             final String defaultEncoding) {
+        this((Run<?, ?>) build, history, result, defaultEncoding);
+
+    }
+
+    /**
+     * Creates a new instance of {@link AggregatedWarningsResult}.
+     *
+     * @param build
+     *            the current build as owner of this action
+     * @param history
+     *            build history
+     * @param result
+     *            the parsed result with all annotations
+     * @param defaultEncoding
+     *            the default encoding to be used when reading and parsing files
+     */
+    public AggregatedWarningsResult(final Run<?, ?> build, final BuildHistory history, final ParserResult result,
```

Comment:
Is this constructor overload actually needed? The only diff I can see is a change of type on the first arg and I'd have thought that was compatible since `AbstractBuild ` is a `Run`.